### PR TITLE
Add bitmap agg type and udaf

### PR DESCRIPTION
**Pull request by kangkaisen**
**1 why remove aligned_free method in port.h?**

it's conflict with RoaringBitmap portability.h.

**2 why remove merge method and add a consume method?**

Initially, I wanted use init and update method instead of consume method,
but I found the logic for key column and raw value column is strange:
what we should we in init and update method for key column and raw value column is not clear.
So I use consume method instead of init and update. the reason is:

	1 the logic is clear: consume the raw data into the data structure related with the column. so, for key column, raw value column and simple agg column, we only nedd copy memory, for complex agg column, we convert the raw data to agg data structure.

	2 the coding is easy.

	3 we aggregate the raw data only when data loading, so we could use consume replace update directly

**3 why not use AggregateInfo::init method?**

	1 when data loading, we use AggregateInfo::consume

	2 when data query and compaction, we use Field::init

**4 why not unify the AggregateInfo::consume and Field::init?**

	AggregateInfo::consume is convert the raw data into agg data structure.

	Field::init is deserialize the binary agg data.

**5 why use Filed instead of ColumnSchema？**

I think Filed content and meaning has contained the  ColumnSchema.

### CODE REVIEW

File: be/src/common/bitmap.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (imay) [inline be/src/common/bitmap.h:1]: be/src/util is better

File: gensrc/thrift/Types.thrift [hunk 2]
```diff
@@ -129,6 +129,7 @@ enum TAggregationType {
     MIN,
     REPLACE,
     HLL_UNION,
+    BITMAP_COUNT,
```
Comment (imay) [inline gensrc/thrift/Types.thrift:132]: I'm afraid that changing NONE's ordinal will lead FE/BE meta corruption

File: be/src/olap/aggregate_func.h [hunk 3]
```diff
@@ -86,6 +92,10 @@ class AggregateInfo {
 };
 
 struct BaseAggregateFuncs {
+    // Default consume do nothing.
+    static void consume(char* dst, const char* src, Arena* arena) {
```
Comment (imay) [inline be/src/olap/aggregate_func.h:96]: 这里为啥不像之前merge那样是nullptr。我理解大部分的cosume应该是跟update相同的，可以直接设置成update的指针。这样还写了不少的consume函数？

File: be/src/olap/aggregate_func.h [hunk 4]
```diff
@@ -336,6 +404,66 @@ struct AggregateFuncTraits<OLAP_FIELD_AGGREGATION_HLL_UNION, OLAP_FIELD_TYPE_HLL
     }
 };
 
+template <>
+struct AggregateFuncTraits<OLAP_FIELD_AGGREGATION_BITMAP_COUNT, OLAP_FIELD_TYPE_VARCHAR> : public BaseAggregateFuncs {
+    static void consume(char* dst, const char* value, Arena* arena) {
+        auto* src = reinterpret_cast<const StringValue*>(value);
+        auto* dest_slice = (Slice*)(dst);
+        char* mem = arena->Allocate(sizeof(RoaringBitmap));
+        new (mem) RoaringBitmap(std::stoi(src->to_string()));
```
Comment (imay) [inline be/src/olap/aggregate_func.h:413]: 我觉得你可以直接new 不需要这种placement new。这种方式你还要显示的调用析构函数的。要不然会有内存泄露的

File: be/src/olap/field_info.cpp [hunk 5]
```diff
@@ -95,6 +95,8 @@ FieldAggregationMethod FieldInfo::get_aggregation_type_by_string(const string& s
         aggregation_type = OLAP_FIELD_AGGREGATION_REPLACE;
     } else if (0 == upper_str.compare("HLL_UNION")) {
         aggregation_type = OLAP_FIELD_AGGREGATION_HLL_UNION;
+    } else if (0 == upper_str.compare("BITMAP_COUNT")) {
```
Comment (imay) [inline be/src/olap/field_info.cpp:98]: 我觉得 bitmap_count 的名字不好，叫做bitmap_union好一些。我们现在是做并集，我理解其实也是可以建立个做交集的

File: be/src/common/bitmap.h (hunk 1 above)
Comment (kangkaisen) [reply be/src/common/bitmap.h:1]: OK

File: gensrc/thrift/Types.thrift (hunk 2 above)
Comment (kangkaisen) [reply gensrc/thrift/Types.thrift:132]: OK. I will keep the NONE's ordinal.

File: be/src/olap/field_info.cpp (hunk 5 above)
Comment (kangkaisen) [reply be/src/olap/field_info.cpp:98]: OK

**Pull request closed without merging**